twitter/searcher.py

```diff
@@ -0,0 +1,182 @@
+            # submissions is a list so we can handle single tweet with multiple media objects
+            submissions, processed_id_tweets = convert_tweet(tweet, processed_id_tweets)
+            if not submissions:
+                continue
+            bulk_upload_submissions(submissions, TWITTER_LARAVEL_API_KEY)
+    print("total_returned_tweets", total_returned_tweets)
+    log_last_processed_id(max_processed_id, max_processed_time_stamp)
+    return max_processed_id
+
+
+def build_tweepy_api():
+    auth = tweepy.OAuthHandler(TWITTER_CONSUMER_KEY, TWITTER_CONSUMER_SECRET)
+    auth.set_access_token(TWITTER_ACCESS_TOKEN, TWITTER_ACCESS_TOKEN_SECRET)
+    return tweepy.API(auth, wait_on_rate_limit=True)
+
+
+def query_twitter(api, query: str, since_id: int):
+    # TODO: include result_type? (mixed, recent, popular) probably want recent once batching is set up
+    # TODO: why wasn't count working
+    return tweepy.Cursor(api.search, q=query, lang="en", since_id=since_id, include_entities=True).items()
+
+
+def convert_tweet(tweet: Dict[str, Any], processed_id_tweets: Set[int]) -> Tuple[List[RawSubmission], Set[int]]:
+    submissions = []
+    # TODO: add "monetizable" logger. like, is somebody profiting off this shit?
+    if not is_tweet_relevant(tweet):
+        return None, processed_id_tweets
+    if "retweeted_status" in tweet:
+        tweet = tweet["retweeted_status"]
+    id_tweet = tweet["id"]
+    if id_tweet in processed_id_tweets:
+        return None, processed_id_tweets
+    processed_id_tweets.add(id_tweet)
+
+    converted_date = twitter_created_at_to_utc(tweet["created_at"])
+    submission_body = ""
+    if tweet["geo"]:
+        submission_body = f"Provided geo: {tweet['geo']}"
+    elif tweet["user"].get("location"):
+        submission_body = f"User Location: {tweet['user']['location']}"
+
+    for media in tweet["extended_entities"]["media"]:
+        if media["type"] != "video":
+            continue
+        print("found video for tweet: ", tweet["text"])
+        media_url = get_media_url_from_variants(media["video_info"]["variants"])
+        if not media_url:
+            continue
+
+        submission = RawSubmission(
+            data_source=DataSource.twitter,
+            id_source=id_tweet,
+            submission_title=tweet["text"],
+            submission_datetime_utc=converted_date,
+            submission_community="",  # TODO: is this ok? I
+            submission_url=media["expanded_url"],
+            submission_media_url=media_url,
+            submission_body=submission_body,
+            id_submitter=tweet["user"]["id"],
+        )
+        submissions.append(submission)
+    return submissions, processed_id_tweets
+
+
+def is_tweet_relevant(tweet: Dict[str, Any]):
+    media = tweet.get("extended_entities", {}).get("media")
+    if media:
+        return True
+    # TODO: do we need a regex like reddit or is that handled by search params
```

Comment:
I think search params are enough. I added the regex because subs like "PublicFreakout" aren't specifically targeted to this stuff

Reply:
cool. thanks